pkg/apis/core/validation/validation.go

```diff
@@ -4294,6 +4303,71 @@ func ValidatePodSpec(spec *core.PodSpec, podMeta *metav1.ObjectMeta, fldPath *fi
 	return allErrs
 }
 
+func validatePodResources(spec *core.PodSpec, podClaimNames sets.Set[string], fldPath *field.Path, opts PodValidationOptions) field.ErrorList {
+	if spec.Resources == nil {
+		return nil
+	}
+
+	allErrs := field.ErrorList{}
+
+	// ValidateResourceRequirements checks if resource names and quantities are
+	// valid, and requests are less than limits.
+	allErrs = append(allErrs, ValidateResourceRequirements(spec.Resources, true, podClaimNames, fldPath, opts)...)
+	allErrs = append(allErrs, validatePodResourceConsistency(spec, fldPath)...)
+	return allErrs
+}
+
+// validatePodResourceConsistency checks if effective container-level requests are
+// less than or equal to pod-level requests, and individual container-level limits
+// are less than or equal to pod-level limits.
+func validatePodResourceConsistency(spec *core.PodSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Convert the *core.PodSpec to *v1.PodSpec to satisfy the call to
+	// resourcehelper.PodRequests method, in the subsequent lines,
+	// which requires a *v1.Pod object (containing a *v1.PodSpec).
+	v1PodSpec := &v1.PodSpec{}
```

Comment:
@jpbetz is this conversion likely to be problematic?  If we need to, we can reimplement `AggregateContainerRequests` against internal (maintaining two copies of these sorts of things is the #1 reason I think we should get rid of "internal")

@ndixita this came up in a different conversation that conversions CAN be expensive, and PodSpec is one of the larger resources.

Reply:
Yes, pods are write heavy and in the critical path of scale. This looks risky to me.  It's safest to assume that it is that it's problematic unless proven otherwise.  A scale test run that sig-scalability signs off on would be my bar.

I'd probably try to find some way to introduce a `resourcehelper.AggregateContainerRequests` variation that works on the internal type.

If this is not fixed now, I'd probably add "either scale test or optimize away conversion" to the beta graduation of this enhancements KEP.

> maintaining two copies of these sorts of things is the #1 reason I think we should get rid of "internal"

+100, I'd strongly prefer we pick a hub type and not maintain an extra internal type. That is, a type that all other types convert through and performs all the roles currently performed by the internal type.

Reply:
So we can maintain a copy of this for internal type. I have 2 concerns:
1. I couldn't find an IFTTT example in the code that will help make sure all the copies of same code are updated together.
2. pull-kubernetes-verify https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/128407/pull-kubernetes-verify/1853996449751109632 forbids  importing k8s.io/component-helpers/resource in  k8s.io/kubernetes/pkg/apis/core/v1: So does that mean we need 3 copies of the code?

Reply:
@jpbetz @thockin ^